app/src/main/java/com/github/se/wanderpals/ui/screens/trip/finance/CurrencySelectionDialog.kt

```diff
@@ -0,0 +1,169 @@
+import androidx.compose.material3.Button
+import androidx.compose.material3.ButtonDefaults
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.OutlinedTextField
+import androidx.compose.material3.Surface
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextFieldDefaults
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.window.Dialog
+import com.github.se.wanderpals.model.viewmodel.FinanceViewModel
+
+/**
+ * A composable function that displays a dialog for currency selection. The user can search the
+ * currency among a list to update the currency used for the trip
+ *
+ * @param financeViewModel The [FinanceViewModel] used to manage the currency selection state.
+ */
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun CurrencySelectionDialog(financeViewModel: FinanceViewModel) {
+
+  var searchedCurrency by remember { mutableStateOf("") }
+  var isError by remember { mutableStateOf(false) }
+  val currencies = Currency.getAvailableCurrencies().filterNot { it.displayName.contains("(") }
+
+  Dialog(
+      onDismissRequest = {
+        isError = false
+        financeViewModel.setShowCurrencyDialogState(false)
+      }) {
+        Surface(
+            modifier =
+                Modifier.fillMaxSize()
+                    .padding(top = 100.dp, bottom = 100.dp)
+                    .testTag("currencyDialog"),
+            color = MaterialTheme.colorScheme.background,
+        ) {
+          Column(
+              modifier = Modifier.fillMaxSize(),
+              horizontalAlignment = Alignment.CenterHorizontally,
+              verticalArrangement = Arrangement.Top) {
+                // Text allowing user to input text and perform search
+                OutlinedTextField(
+                    value = searchedCurrency,
+                    onValueChange = { value -> searchedCurrency = value },
+                    modifier =
+                        Modifier.fillMaxWidth()
+                            .padding(20.dp)
+                            .clip(RoundedCornerShape(5.dp))
+                            .testTag("currencySearchText"),
+                    label = {
+                      Text(
+                          text = if (isError) "Invalid currency" else "Select a currency",
```

Comment:
Could have used when then were instead of if else to reduce the cognitive complexity. From my understanding, when then doesn't count but if else does

Reply:
Here using when seems a bit overkilled since we have just an if else case